generator/src/test/java/com/scottlogic/deg/generator/inputs/profileviolation/ProfileViolationTests.java

```diff
@@ -0,0 +1,126 @@
+package com.scottlogic.deg.generator.inputs.profileviolation;
+
+import com.scottlogic.deg.generator.Field;
+import com.scottlogic.deg.generator.Profile;
+import com.scottlogic.deg.generator.ProfileFields;
+import com.scottlogic.deg.generator.Rule;
+import com.scottlogic.deg.generator.builders.AndBuilder;
+import com.scottlogic.deg.generator.builders.OrBuilder;
+import com.scottlogic.deg.generator.builders.RuleBuilder;
+import com.scottlogic.deg.generator.outputs.manifest.ManifestWriter;
+import com.scottlogic.deg.generator.violations.ViolatedProfile;
+import com.scottlogic.deg.generator.violations.filters.ViolationFilter;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.*;
+
+import static com.scottlogic.deg.generator.inputs.profileviolation.TypeEqualityHelper.assertListProfileTypeEquality;
+import static com.shazam.shazamcrest.MatcherAssert.assertThat;
+import static com.shazam.shazamcrest.matcher.Matchers.sameBeanAs;
+import static org.mockito.Matchers.anyListOf;
+import static org.mockito.Mockito.*;
+
+/**
+ * Defines tests for all business logic involved in Profile Violation.
+ */
+public class ProfileViolationTests {
+    private IndividualRuleProfileViolator target;
+    private ArrayList<ViolationFilter> constraintsToNotViolate;
+
+    @Mock private ManifestWriter mockManifestWriter;
+    @Mock private Path mockPath;
+
+    @BeforeEach
+    public void setUp() throws IOException {
+        MockitoAnnotations.initMocks(this);
+        doNothing()
+            .when(mockManifestWriter)
+            .writeManifest(anyListOf(ViolatedProfile.class), same(mockPath));
+
+        constraintsToNotViolate = new ArrayList<>();
+        IndividualConstraintRuleViolator ruleViolator = new IndividualConstraintRuleViolator(constraintsToNotViolate);
+        target = new IndividualRuleProfileViolator(mockManifestWriter, mockPath, ruleViolator);
+    }
+
+    /**
+     * Tests that the violator can take a profile
+     *  Input: Profile with 2 fields foo and bar, 2 single atomic constraint rules affecting foo and bar
+     *  Output: 2 Profiles, one with rule 1 negated and rule 2 unaffected, one with rule 1 unaffected and rule 2 negated
+     */
+    @Test
+    public void violate_withTwoSimpleRuleProfile_producesTwoViolatedProfiles() throws IOException {
+        //Arrange
+        Field fooField = new Field("foo");
+        Rule rule1 = new RuleBuilder("Rule 1")
```

Comment:
Is there a way to make this test more concise? This one test is ~100 lines long, presumably there would be lots of tests here making this easily a 1000+ line test fixture?

Reply:
Won't fix now but will bear this in mind when adding future scenarios